File: 2015/Day 25/day25.cpp
//
//  Advent of Code 2015
//  Day 25
//
//  By PatchesTheDipstick
//


//  System includes...

#include <regex>

//  Third Party includes...



//  Project includes...

#include "../utils.hpp"


int main(int argc, char** argv)
{
	std::vector<std::string> input = ptd::utils::getInput("input.txt");
	
	std::regex pattern (".*row ([0-9]*), column ([0-9]*).");
	std::smatch token;
	std::regex_match(input[0], token, pattern);
	long long int row = std::stoi(token[1]);
	long long int col = std::stoi(token[2]);
	long long int last_row = row + col - 1;
	long long int index = 0;
	for (long long int a = 1; a <= last_row - 1; a++)
		index += a;
	index += col;
	long long int value = 20151125;
	for (long long int a = 2; a <= index; a++)
	{
		value = (value * 252533) % 33554393;
		
	}

	
	std::cout << "Advent of Code 2015 Day  answers:" << std::endl;
	std::cout << "    Part 1 : " << value << std::endl;
	std::cout << "    Part 2 : " << "Free Star" << std::endl;

	return (0);

}

File: 2015/Day 25/day25.py
#
#  Advent of Code 2015
#  Day 
#
#  By PatchesTheDipstick
#
import re

def getInput(file_name):
	input = []
	with open(file_name) as input_file:
		input = input_file.read().splitlines()
	return input

if __name__ == '__main__':
	
	input = getInput("input.txt")
	
	pattern = re.compile('.*row (\d*), column (\d*).')
	token = pattern.search(input[0])
	row = int(token.group(1))
	col = int(token.group(2))
	
	last_row = row + col - 1
	index = 0
	for a in range(1, last_row):
		index += a
	index += col
	print(index)
	grid = []
	value = 20151125
	for a in range(2, index + 1):
		print(a)
		value *= 252533
		value %= 33554393
	
	
	
	part_one = value
	part_two = 0
	print("Advent of Code 2015 Day 03")
	print("  Part one:", part_one)
	print("  Part two: Free Star")
	